utils/dwarf.c

```diff
@@ -1430,9 +1431,22 @@ static void get_source_location(Dwarf_Die *die, struct build_data *bd,
 		Dwarf_Die cudie;
 		Dwarf_Line *line;
 		unsigned long dwarf_addr = sym_to_dwarf_addr(dinfo, sym->addr);
+		unsigned long limit_addr = dwarf_addr + sym->size;
 
 		dwarf_diecu(die, &cudie, NULL, NULL);
-		line = dwarf_getsrc_die(&cudie, dwarf_addr);
+
+		/*
+		 * This loop is needed because gcc doesn't create dwarf info at NOP
+		 * instruction address so dwarf_getsrc_die() returns NULL.
+		 * To avoid this problem, we move forward until we see the first actual
+		 * instruction address of the function, which has a valid dwarf info.
+		 * The search is limited to the function size.
+		 */
+		do {
+			line = dwarf_getsrc_die(&cudie, dwarf_addr);
+			dwarf_addr += NOP_INSN_SIZE;
+		} while (line == NULL && dwarf_addr < limit_addr);
```

Comment:
Can we check both the function size and the count?  I'm afraid if there's a pathological case..

Reply:
What could be the count? Do you mean we can set an arbitrary count limit just like `--search_limit > 0`?

Reply:
Yes, I meant that.

Reply:
OK. Then I will set `int search_limit = 10` and check the count.

Reply:
I just update it. Thanks.